### ISSUE TITLE
MAME 2014 links are down??

### ISSUE THREAD

**Issue by NerdFlix** [zach-morris/plugin.program.iarl#35]
Hi zach Morris, firstly I wanna say thank you for your hard work and effort to give us this amazing addon.
Some games like Street fighter 2 on Mame 2014 list arent downloading, as well as the archive neogeo.zip, I think not all of them are in archive.org anymore, is that true?
I tried the direct link of some of them and cant find the file to download(in MAME_2014_ZachMorris.xml)

LINK: https://archive.org/download/MAME0.159_MAME2014_Reference_Set_ROMs_CHDs_Samples/roms/

**Comment by zach-morris** [comment]
They all seem to be working for me.  The nice thing about archive.org is the lists are generally pretty static (although the Final Burn Alpha one changed somewhat recently).  Debug log?

**Comment by NerdFlix** [comment]
Can you please download and play for example the game Street Fighter II: The World Warrior (World 910522) in MAME_2014_ZachMorris.xml??, direct link: http://archive.org/download/MAME0.159_MAME2014_Reference_Set_ROMs_CHDs_Samples/roms/sf2.zip
I can download and play just fine for instance the game Mortal Kombat 3 (rev 2.1) in the same xml, direct link: http://archive.org/download/MAME0.159_MAME2014_Reference_Set_ROMs_CHDs_Samples/roms/mk3.zip
If you open the street fighter 2 link in your browser you will see that there is nothing there.
The games that depend on the file neogeo.zip won't run because that file isn't there too, direct link: http://archive.org/download/MAME0.159_MAME2014_Reference_Set_ROMs_CHDs_Samples/roms/neogeo.zip
Thanks in advance.

**Comment by zach-morris** [comment]
``DEBUG: IARL:  Download with login URL: http://archive.org/download/MAME0.159_MAME2014_Reference_Set_ROMs_CHDs_Samples/roms/sf2.zip
19:36:04.195 T:123145303453696   DEBUG: IARL:  Download save filename: /Users/ZC/Library/Application Support/Kodi/userdata/addon_data/plugin.program.iarl/temp_iarl/sf2.zip``

**Comment by zach-morris** [comment]
Well, I take this back.  I tried again and now it's not working.  I'm assuming there's something going on with either the archive or the site itself.  I'll leave this open for now and see if it corrects itself.  If not I'll look for an alternate

**Comment by zach-morris** [comment]
Added [this commit](https://github.com/zach-morris/plugin.program.iarl/commit/c50915f9ec5557bce082b4bc86d0ae49be00256e) to help identify no longer available files.

**Comment by NerdFlix** [comment]
Yeah, some games that depend on QSound.zip or neogeo.zip won't run because both files aren't there. Well I still think that games are not downloading because they simple don't exist there anymore.
Anyway thanks for answering  :)

**Comment by zach-morris** [comment]
You can also try the normal MAME list, which I believe is ver .172.  You can still use MAME 2014 core, but it working will depend on if the rom changed between .158 and .172.  Another option is to use MAME 2003 or MAME 2016 with the .172 list.  Like I said, if the archive isn't restored in a while, I'll look for an alternate source (which there usually are thankfully).

**Comment by zach-morris** [comment]
I like the torrent idea.  With v18 of Kodi, Quasar might work right out of the box with game torrents

**Comment by NerdFlix** [comment]
Hi zach,
as you said before, "IARL will stick to html statically linkable sources only (focusing on archive.org)", so isn't there a chance to add some roms from other sources like Emuparadise? I took a look at their site and for MAME 2014 games, almost all of them are there, I tested here and using an already made emuparadise scraper, it is too easy to get the download links and they are pretty static, you just have to build a custom header before making the http request. I've created an xml list for gamecube and it is working well, downloading games and showing videos and images(I've scraped them from TheGamesDB). Basically it is possible to offer games from almost any platform presented in emuparadise. Let me know if you are interested :)

**Comment by zach-morris** [comment]
MAME 2014 is currently being updated on archive.org, so I imagine this will be resolved soon.  However, I'll leave this open for an enhancement:  Post DL action to 'Rollback' to a specific MAME version.  MAME 0.185 romset and 0.185 rollback romset are available, so in theory, IARL should be able to combine the required files to create any older MAME romset version using those two sets.

**Comment by NerdFlix** [comment]
This is great news. Let me ask you something, what Retroarch core do you take to run Street fighter 3 roms?
